### DISCUSSION TITLE
Handle products that expire

### DISCUSSION THREAD

**Discussion by tobbbe** [Shopify/storefront-api-feedback#216] [Help]
Hello!
Is it possible to set an expiration date for products and then filter them using the storefront API? Using metafields seems like a good approach, but I dont think you can filter products by metafields of datetime? Can tags and the query be used somehow to do what I want?

If its not possible with storefront api, any other suggestion would be very appreciated! 😊

**Comment by johncraigcole** [comment]
Hey @tobbbe this for a Shopify plus merchant?

The Storefront API metafield querying is limited to an equality check. If you defined a metafield as a date, you couldn't do something like get me all the products who's expiry was greater than today. If we're talking about a plus store, I was wondering about just removing it from the app channel using the Admin API [publishableUnpublish](https://shopify.dev/docs/api/admin-graphql/2023-07/mutations/publishableunpublish) mutation, as anything not available to the app won't be returned.

A general products query could do a product doesn't have tag type query. For example, return first 100 products that aren't marked with a tag expired.

```
	products(first:100 query:"-tag:'expired'"){
		nodes{
			title
			tags
		}
	}
```

Metafield filtering is limited to in [collection products connection](https://shopify.dev/docs/api/storefront/2023-07/objects/Collection#connection-collection-products), and the [search](https://shopify.dev/docs/api/storefront/2023-07/queries/search#argument-search-productfilters) query.

**Comment by tobbbe** [comment]
Hey! Thanks for the reply 😃
Unfortunately its not a Plus store. Does publishableUnpublish (and also publishablePublish) require Plus btw? I cant find that they do anywhere in the docs ([here for example](https://shopify.dev/docs/api/usage/access-scopes#authenticated-access-scopes)). If they do Im in trouble 😅

Is there any plans to add any other filters to metafields? Date comparison, gte, lte would be great for example.

Checking tags could be an idea. But that would require us to add the tag to all products. Is it possible to do have something like named tags, ex expiresat:20230220 and filter based on that?

My last resort would be to use the Admin api and run a check on all products every day and then set an "expire" tag. Would that be possible?

**Reply by johncraigcole** [reply]
At one point `publishablePublish` was restricted to Plus only stores but it's been changed and I wasn't aware of that change. You're not in trouble. 😄 

No timeline on when the metafield type filtering will get greater search operator flexibility like gte,lte etc.

The problem I see with the named tag vs. something like `expired` is that I'm assuming the `expiresat` isn't going to be the same date across all products so I'd just prefer to just say in my query to only give me products that are marked as expired. Yes, you could have a job that runs daily to say show me all products where `expiresat:20231011` and add the `expires` tag.
